Sources/Diffing.swift

```diff
@@ -40,3 +41,104 @@ public extension DiffableViewModel {
         return String(describing: Self.self)
     }
 }
+
+/// MARK: - DifferenceKit Helpers
+
+/// Creates a type-erased Differentiable for DiffableViewModel.
+/// These are only created internally from either `TableCellViewModel` or `CollectionCellViewModel`,
+/// so that we can safely force cast the models back to those types.
+public struct AnyDiffableViewModel {
+    public let model: DiffableViewModel
+
+    private let isContentEqualTo: (AnyDiffableViewModel) -> Bool
+
+    init(_ model: TableCellViewModel) {
+        self.model = model
+
+        let differenceIdentifier = model.diffingKey
+        /// Only compares diff identifiers. This means we'll never get "reload"-type `Changeset`s
```

Comment:
hm... isn't this a problem? what if a cell updates it's content and needs to be refreshed? (but it's diff identifier wouldn't change...)

Reply:
We do it manually for the cells on screen in ReactiveLists, which is how it works today.

Reply:
We could revisit this in the future to see if this is something we want to support (using diffing to animate reloading of cells), but right now it would break forms (e.g. jumping fields because of keyboard dismissal)

Reply:
ah, right 👍 sounds good to me